Factory/Config/StackFactory.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+/*
+ * This file is part of the `liip/LiipImagineBundle` project.
+ *
+ * (c) https://github.com/liip/LiipImagineBundle/graphs/contributors
+ *
+ * For the full copyright and license information, please view the LICENSE.md
+ * file that was distributed with this source code.
+ */
+
+namespace Liip\ImagineBundle\Factory\Config;
+
+use Liip\ImagineBundle\Config\Stack;
+
+class StackFactory
+{
+    /**
+     * @param string      $name
+     * @param string|null $dataLoader
+     * @param int|null    $quality
+     * @param array       $filters
+     *
+     * @return Stack
+     */
+    public function create(string $name, string $dataLoader = null, int $quality = null, array $filters)
+    {
+        return new Stack($name, $dataLoader, $quality, $filters);
```

Comment:
also, is there value to use a factory for this over just doing `new Stack()` in the place where we need the stack? feels to me like the factory is too much abstraction.

Reply:
using `new` is breaking DI principle. it makes it hardcoded and non-extendable and not testable.

Reply:
i don't see the problem. we don't want a mock stack anywhere, its a value object. to me it looks overengineered with this. we do not need to DI what kind of stack is created as its just a value object. but its not a huge deal either.

Reply:
we already mock it in our unit tests

Reply:
i don't see the need for mocking the Stack object in a unit test, its straightforward enough imho. yes, its not a perfectly atomic unit test, but good enough as all of the build process is just infrastructure code to build the configuration object.